jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/GradleLayerConfigurations.java

```diff
@@ -137,7 +137,11 @@ private static JavaLayerConfigurations getForNonWarProject(
         layerBuilder.addFile(
             layerType,
             dependencyFile.toPath(),
-            dependenciesExtractionPath.resolve(dependencyFile.getName()));
+            dependenciesExtractionPath.resolve(
+                dependencyFile.getName().substring(0, dependencyFile.getName().length() - 4)
+                    + "-"
+                    + Files.size(dependencyFile.toPath())
```

Comment:
I don't know where that comment went about duplicates, but streams has a pretty fun way to calculate them.

```
    List<String> duplicates = paths
        .stream()
        .map(Path::getFileName)
        .map(Path::toString)
        .collect(Collectors.groupingBy(filaname -> filename, Collectors.counting()))
        .entrySet()
        .stream()
        .filter(entry -> entry.getValue() > 1)
        .map(Entry::getKey)
        .collect(Collectors.toList());
```

and then apply the tranformation after checking the duplicates list?
```
duplicates.contains(dependencyFile.getName()) ? transformName : originalName
```

Reply:
Ah ok, nice.